pg_analytics/src/datafusion/context.rs

```diff
@@ -171,23 +172,74 @@ impl ParadeContextProvider {
             options: ConfigOptions::new(),
         })
     }
+
+    fn get_table_source_impl(
+        reference: TableReference,
+    ) -> Result<Arc<dyn TableSource>, ParadeError> {
+        let table_name = reference.table();
+        let schema_name = reference.schema();
+
+        if let Some(schema_name) = schema_name {
+            // If a schema was provided in the query, i.e. SELECT * FROM <schema>.<table>
+            DatafusionContext::with_schema_provider(schema_name, |provider| {
+                let table = task::block_on(provider.table(table_name))
+                    .ok_or(NotFound::Table(table_name.to_string()))?;
+
+                Ok(provider_as_source(table))
+            })
+        } else {
+            // If no schema was provided in the query, i.e. SELECT * FROM <table>
+            // Read all schemas from the Postgres search path and cascade through them
+            // until a table is found
+            let current_schemas = unsafe {
+                direct_function_call::<Array<pg_sys::Datum>>(
+                    pg_sys::current_schemas,
+                    &[Some(pg_sys::Datum::from(true))],
+                )
+            };
+
+            if let Some(current_schemas) = current_schemas {
+                for datum in current_schemas.iter().flatten() {
```

Comment:
It seems like you're flattening out `None` values here. Why would a schema be `None`? Just want to make sure we're not missing a case that needs to be handled.

Reply:
I don't think Postgres would ever return a null schema; `T` of `pgrx::Array<T>` is an `Option`. Unless I'm missing something?